Cortex 10 cleanup of simple \deprecated

andrewkaufman opened the pull request:
This PR is a mish-mash of all the easy \deprecated stuff that's been hanging around. There are 3 remaining \deprecated tags, all todo with IECorePython Wrappers, which I'll handle in a separate PR (since I assume they'll result in a large diff).

I've run `scons test`, `scons testGL`, `scons testRI`, `scons testArnold`, and `scons testMaya`, and while they don't all pass, they all have the same failures as on 9_maintenance.

andrewkaufman commented:
Some points of interest:

- I cheated for ImagePrimitive constructor (5942c454ec1d63313e0514d5df357b35a171fc1f)
- I wasn't sure if throw or warn or assert was the right thing here (https://github.com/ImageEngine/cortex/pull/581/commits/612b0473606c5020e7ee543bbfb4b0a020e99f1e#diff-6be15841e6994769628b8e5e49683d17R112)
- Not sure about the default arg here, but seems a read-only UniverseBlock is a decent default, and I was having trouble getting the bindings to compile otherwise (12335ce8de2151141bb1cf653435b00ffa047569)
  - Also, I haven't cleaned up a `ClassData` in a while, but I think I did the right thing anyways.
- Note sure we actually need to access `_topLevelUI()` here, but that keeps things as they were before (https://github.com/ImageEngine/cortex/pull/581/commits/3d5d54e05c6c9a3549d3d336aa12bf9012cbbce7#diff-9b2f67ab468efb8b6b828e909d3c78cbR71)
  - Also, I haven't tested interactively in Maya yet, just the tests. Its quite a bit of IE code would need compiling before interactive Maya would be testable easily.

andrewkaufman commented:
I added 3 new commits for deprecations that weren't marked with \deprecated, but rather with a note saying "change for next major version". I did have to force push after a rebase on master, but the only changes are 	a406992, 0b57952, and a65bdf1.

johnhaddon commented on src/IECoreGL/MeshPrimitive.cpp:
```diff
@@ -139,28 +96,26 @@ void MeshPrimitive::addPrimitiveVariable( const std::string &name, const IECore:
 	}
+	else if ( primVar.interpolation==IECore::PrimitiveVariable::Vertex || primVar.interpolation==IECore::PrimitiveVariable::Varying )
+	{
+		// we should never get here, because ToGLMeshConverter should have already
```
We could definitely get here, because `addPrimitiveVariable()` is a public method that could be called by things other than ToGLMeshConverter. I think throwing is the appropriate response, but the comment is a bit misleading - maybe just remove it?

johnhaddon commented on include/IECoreGL/Renderer.h:
```diff
@@ -449,17 +449,11 @@ class IECOREGL_API Renderer : public IECore::Renderer
-		///	"thetaMax"	FloatData	360
-		///
-		/// \deprecated Use the sphere() method instead.
+		/// Implemented as a pass-through
```
Pass-through to what? Aren't these basically just unimplemented?

johnhaddon commented on contrib/IECoreArnold/include/IECoreArnold/UniverseBlock.h:
```diff
@@ -50,13 +50,11 @@ class IECOREARNOLD_API UniverseBlock : public boost::noncopyable
 		/// have been loaded. If writable is true, then throws if
 		/// there is already a writer.
-		UniverseBlock( bool writable );
+		UniverseBlock( bool writable = false );
```
I think I'd prefer not having a default argument here, and forcing the caller to think about what they need.

andrewkaufman commented:
I forced these in because they seemed simple enough. The only commits that have changed are:

- d920a88 : remove comment
- b37ecaa : changed comments to say "Not Implemented"
- ca9a37c : removed default arg for constructor and fixed binding

andrewkaufman commented:
Rebased with the UniverseBlock py/test fix squashed into 189739a

andrewkaufman commented:
I updated the UniverseBlock commit one more time (803192c) to remove the ClassData include, and I've added one more commit to remove ClassData from a comment (8078fb7). With this, the only use of ClassData is in its own .h/.inl files.

johnhaddon commented:
Nice one - thanks!
